x/pocketcore/keeper/proof.go

```diff
@@ -50,13 +50,21 @@ func (k Keeper) SendProofTx(ctx sdk.Ctx, n client.Client, node *pc.PocketNode, p
 				ctx.Logger().Error(fmt.Sprintf("could not delete evidence is not sealed, could cause a relay leak: %s", err.Error()))
 			}
 		}
-		if evidence.NumOfProofs != claim.TotalProofs {
+
+		if evidence.NumOfProofs < claim.TotalProofs {
+			//  The node should never submit a proof whenever the number of stored relay proofs are
+			//  LESS than the number of relay proofs in the submitted claim as it can cause
+			//  a array index of out-bounds exception if the randomly selected index is more than what
+			//  we have in the store, causing a process crash.
 			err := pc.DeleteEvidence(claim.SessionHeader, claim.EvidenceType, node.EvidenceStore)
 			ctx.Logger().Error(fmt.Sprintf("evidence num of proofs does not equal claim total proofs... possible relay leak"))
```

Comment:
Can you update the error message as well?

Reply:
Sounds good, done